### ISSUE TITLE
MongoDB BigInt query

### ISSUE THREAD

**Issue by neo-luca** [DataDog/dd-trace-js#3548]
**Expected behaviour**
DataDog tracer should be able to handle without any problem MongoDB queries that includes BigInt.

**Actual behaviour**
`dd-trace` throws the following error every time it tries to log a MongoDB query that includes BigInt.
`TypeError: Do not know how to serialize a BigInt`
`node_modules/dd-trace/packages/datadog-plugin-mongodb-core/src/index.js:42`
`if (cmd.filter) return JSON.stringify(limitDepth(cmd.filter))`

**Steps to reproduce**

`package.json`
```json
{
  "name": "mongodb-dd-trace-bigint",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "dd-trace": "^4.12.0",
    "mongodb": "^5.7.0"
  }
}
```

`index.js`
```javascript
const tracer = require('dd-trace').init();
const { MongoClient } = require('mongodb');

(async () => {
    const client = new MongoClient('mongodb://localhost:27017');
    await client.connect();
    const db = client.db('test');
    const collection = db.collection('transactions');
    const transactions = await collection.find({
        amountCents: BigInt(12)
    }).toArray();
    console.log(transactions.length);
    await client.close();
})();
```

**Environment**

* **Operation system:** macOS Ventura 13.3.1
* **Node.js version:** 18.15.0
* **Tracer version:** 4.12.0
* **Agent version:** n/a
* **Relevant library versions:** n/a

**Comment by amjad-mahfoud-neo** [comment]
we can change the code in [the getQuery function](https://github.com/amjad-mahfoud-neo/dd-trace-js/blob/4cb1288f0e15ebe3dd1a676ec8a61f52b20c07f1/packages/datadog-plugin-mongodb-core/src/index.js#L39) to be something like this
```javascript
function sanitizeBigInt (data) {
  return JSON.stringify(data, (key, value) => typeof value === 'bigint' ? value.toString() : value)
}

function getQuery (cmd) {
  if (!cmd || typeof cmd !== 'object' || Array.isArray(cmd)) return
  if (cmd.query) return sanitizeBigInt(limitDepth(cmd.query))
  if (cmd.filter) return sanitizeBigInt(limitDepth(cmd.filter))
}

```

**Comment by tlhunter** [comment]
This fix should be available in v4.14.0 and v3.35.0.
